projectorganizer/src/prjorg-project.c

```diff
@@ -658,6 +688,36 @@ GtkWidget *prjorg_project_add_properties_tab(GtkWidget *notebook)
 	gtk_box_pack_start(GTK_BOX(hbox1), label, FALSE, FALSE, 12);
 	gtk_box_pack_start(GTK_BOX(vbox), hbox1, FALSE, FALSE, 0);
 
+	hbox1 = gtk_hbox_new(FALSE, 0);
+	label = gtk_label_new(NULL);
+	gtk_label_set_markup(GTK_LABEL(label), _("<b>External Programs</b>"));
+	gtk_box_pack_start(GTK_BOX(hbox1), label, FALSE, FALSE, 0);
+	gtk_box_pack_start(GTK_BOX(vbox), hbox1, FALSE, FALSE, 12);
+
+	table = gtk_table_new(1, 2, FALSE);
+	gtk_table_set_row_spacings(GTK_TABLE(table), 6);
+	gtk_table_set_col_spacings(GTK_TABLE(table), 12);
+
+	label = gtk_label_new(_("File manager:"));
+	gtk_misc_set_alignment(GTK_MISC(label), 0, 0);
+	e->command_file_manager = gtk_entry_new();
+	ui_table_add_row(GTK_TABLE(table), 0, label, e->command_file_manager, NULL);
+	ui_entry_add_clear_icon(GTK_ENTRY(e->command_file_manager));
+	gtk_widget_set_tooltip_text(e->command_file_manager,
+		_("The command used to open folders in the file manager."));
+	gtk_entry_set_text(GTK_ENTRY(e->command_file_manager), prj_org->command_file_manager);
+
+	label = gtk_label_new(_("Terminal:"));
+	gtk_misc_set_alignment(GTK_MISC(label), 0, 0);
+	e->command_terminal = gtk_entry_new();
+	ui_table_add_row(GTK_TABLE(table), 2, label, e->command_terminal, NULL);
+	ui_entry_add_clear_icon(GTK_ENTRY(e->command_terminal));
+	gtk_widget_set_tooltip_text(e->command_terminal,
+		_("The terminal emulator."));
+	gtk_entry_set_text(GTK_ENTRY(e->command_terminal), prj_org->command_terminal);
+
+	gtk_box_pack_start(GTK_BOX(vbox), table, FALSE, FALSE, 6);
+
```

Comment:
I'd really like this to be a plugin configuration instead of a project configuration or not configurable at all. If you don't want to make it part of this pull request, I'd actually prefer if you could remove the configuration code from this pull request.